### ISSUE TITLE
Coded features review: settlement at expiry

### ISSUE THREAD

**Issue by witgaw** [vegaprotocol/specs#561]
See https://github.com/vegaprotocol/specs-internal/blob/master/protocol/0002-settlement.md and other specs linked from it (e.g. https://github.com/vegaprotocol/specs-internal/blob/master/protocol/0016-product-builtin-future.md)

PR for feature tests - https://github.com/vegaprotocol/vega/pull/4212
Raised bug & Blocked by https://github.com/vegaprotocol/vega/issues/4304
Closes Core ticket https://github.com/vegaprotocol/vega/issues/3820

**Comment by gordsport** [comment]
Hi @witgaw,
Its thought that this issue is now resolved and cant be closed. Does this fit with your thinking?

**Comment by davidsiska-vega** [comment]
Blocked by https://github.com/vegaprotocol/vega/pull/4212 and by https://github.com/vegaprotocol/vega/pull/4304 .

**Comment by witgaw** [comment]
Reviewed the test and it's now merged into core.
